### PR TITLE
New homepage design, with A/B testing implementation.

### PR DESCRIPTION
**Pull request by graysonwright**
New homepage design, with A/B testing implementation.

hound

### CODE REVIEW

File: spec/features/landing_page_split_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    expect(page).to have_content(new_landing_page_content)
+    # expect(segment).to have_received(landing_analytics_event(:new))
+  end
+
+  def landing_page(alternative)
+    "/?landing_page=#{alternative}"
+  end
+
+  def original_landing_page_content
+    "Review your JavaScript, CoffeeScript, and Ruby code for style guide violations with a trusty hound."
```
Comment (croaky) [inline spec/features/landing_page_split_spec.rb]: We should probably add SCSS here.

File: app/views/landing/new.haml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+- content_for :page_title, "Hound"
```
Comment (croaky) [inline app/views/landing/new.haml]: How are we going to distinguish between old and new in the Mixpanel funnels? Should this be a different name? Or, send properties through for the event? Maybe Segment has suggestions.

File: spec/features/landing_page_split_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+require "spec_helper"
+
+feature "Landing page split testing" do
+  scenario "user sees original landing page" do
+    visit landing_page(:original)
```
Comment (croaky) [inline spec/features/landing_page_split_spec.rb]: If one version is called "new", for symmetry should we call the other one "old"? Or name them "a" and "b"?

Comment (graysonwright) [inline spec/features/landing_page_split_spec.rb]: I'm waiting to see what Don has for the landing page – I'd like to replace `new` with something descriptive about the new design, like `twitter_reviews` or `banner_image` or `purple_background`.

Comment (graysonwright) [inline spec/features/landing_page_split_spec.rb]: I'd like to be able to re-use this for different landing page experiments.

**Comment by graysonwright** [comment]
To access the new landing page, go to http://localhost:5000/?landing_page=benefits

File: app/views/application/_segment_io.haml [hunk 4]
```diff
@@ -1,7 +1,9 @@
 :javascript
   window.analytics||(window.analytics=[]),window.analytics.methods=["identify","track","trackLink","trackForm","trackClick","trackSubmit","page","pageview","ab","alias","ready","group","on","once","off"],window.analytics.factory=function(t){return function(){var a=Array.prototype.slice.call(arguments);return a.unshift(t),window.analytics.push(a),window.analytics}};for(var i=0;i<window.analytics.methods.length;i++){var method=window.analytics.methods[i];window.analytics[method]=window.analytics.factory(method)}window.analytics.load=function(t){var a=document.createElement("script");a.type="text/javascript",a.async=!0,a.src=("https:"===document.location.protocol?"https://":"http://")+"d2dq2ahtl5zl1z.cloudfront.net/analytics.js/v1/"+t+"/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(a,n)},window.analytics.SNIPPET_VERSION="2.0.8",
   window.analytics.load("#{ENV['SEGMENT_IO_WRITE_KEY']}");
-  window.analytics.page("#{yield :page_title}");
+  window.analytics.page("#{
+    [yield(:page_title), yield(:variation)].compact.uniq.join(" > ")
```
Comment (croaky) [inline app/views/application/_segment_io.haml]: Worth pulling into a view helper? Not sure it's worth it.

File: app/views/landing/original.haml [hunk 5]
```diff
@@ -1,4 +1,5 @@
 - content_for :page_title, "Hound"
+- content_for :body_class, "landing original"
```
Comment (croaky) [inline app/views/landing/original.haml]: Also?

```
- content_for :variation, "original"
```

**Comment by frechg** [comment]
Looking for some early review:

hound

File: app/views/landing/benefits.haml [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    .inner-wrapper
+      %article
+        %figure.media-object
+          .icon
+            %i.fa.fa-bolt
+        .media-body
+          %h4 Convention
+          %p
+            Defining and sticking with a style guide helps reduce the amount of
+            decisions that developers needs to make when writing code, helping to
```
Comment (salbertson) [inline app/views/landing/benefits.haml]: need?

File: app/views/landing/benefits.haml [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+            keep their focus on logic not semantics and formatting.
```
Comment (salbertson) [inline app/views/landing/benefits.haml]: "logic, not..."? I think I overuse commas so feel free to ignore.

File: app/views/landing/benefits.haml [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+        %figure.media-object
+          .icon
+            %i.fa.fa-ellipsis-h
+        .media-body
+          %h4 Consistency
+          %p
+            When every line of code is written in the same style, the whole
+            codebase becomes easier to read, understand and de-bug. It also
+            helps prevent areas of the codebase from being “owned” by one
+            persons unique style vs. the team.
```
Comment (salbertson) [inline app/views/landing/benefits.haml]: person's? Possesive?

File: app/views/landing/benefits.haml [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+%section.configuration
+  .section-content
+    %figure.media-object
+      = image_tag 'guides.png'
+    .media-body
+      %h4
+        Use our guides or configure your own.
+      %p
+        Hounds default style guides build upon accepted industry standards and
```
Comment (salbertson) [inline app/views/landing/benefits.haml]: Hound's?

File: app/views/landing/benefits.haml [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+        are the result of years of utilization in software projects at
+        thoughtbot. Have your own style guide or need to modify the default?
+        Hound provides the ability to customize your guides to best suit your
+        teams needs.
```
Comment (salbertson) [inline app/views/landing/benefits.haml]: team's?

File: spec/features/landing_page_split_spec.rb (hunk 1 above)
Comment (salbertson) [inline spec/features/landing_page_split_spec.rb]: @frechg, do we want SCSS included?

**Comment by salbertson** [comment]
Closing this since we are splitting up the A/B testing and new landing page changes. @graysonwright, feel free to open up another PR for the A/B testing if we need to use it.

**Comment by graysonwright** [comment]
Alright,sounds good

**Pull request closed without merging**